File: euler_python/problem56.py
"""
problem56.py

Considering natural numbers of the form, a**b, where a, b < 100,
what is the maximum digital sum?
"""

def digit_sum(n):
    return sum(map(int, (str(n))))

def problem56():
    return max(digit_sum(a**b) for a in range(100) for b in range(100))

File: euler_clojure/src/euler/problem56.clj
(ns euler.problem56
  "https://projecteuler.net/problem=56

  Considering natural numbers of the form, a**b, where a, b < 100,
  what is the maximum digital sum?"
  (:require [clojure.math.numeric-tower :as math]
            [euler.toolset :refer [digits]]))

(defn digit-sum [n]
  (reduce + (digits n)))

(defn problem56 []
  (apply max (for [a (range 100)
                   b (range 100)]
               (digit-sum (math/expt a b)))))
